discovery-provider/plugins/notifications/src/__tests__/mappings/tastemaker.test.ts

```diff
@@ -0,0 +1,119 @@
+import { expect, jest, test } from '@jest/globals'
+import { renderEmail } from '../../email/notifications/renderEmail'
+import { Processor } from '../../main'
+import * as sns from '../../sns'
+import { AppEmailNotification } from '../../types/notifications'
+
+import {
+  createUsers,
+  insertMobileDevices,
+  insertMobileSettings,
+  insertNotifications,
+  createTracks,
+  setupTest,
+  resetTests
+} from '../../utils/populateDB'
+
+describe('Tastemaker Notification', () => {
+  let processor: Processor
+
+  const sendPushNotificationSpy = jest
+    .spyOn(sns, 'sendPushNotification')
+    .mockImplementation(() => Promise.resolve())
+
+  beforeEach(async () => {
+    const setup = await setupTest()
+    processor = setup.processor
+  })
+
+  afterEach(async () => {
+    await resetTests(processor)
+  })
+
+  test('Process push notification for trending track', async () => {
+    await createUsers(processor.discoveryDB, [
+      { user_id: 1 },
+      { user_id: 2 },
+      { user_id: 3 }
+    ])
+    await createTracks(processor.discoveryDB, [{ track_id: 3, owner_id: 2 }])
+
+    // User 1 follows user 2
+    const notificationRow = {
+      id: 1,
+      specifier: '10',
+      group_id:
+        'tastemaker_user_id:10:tastemaker_item_id:3',
+      type: 'tastemaker',
+      timestamp: new Date(),
+      data: { tastemaker_item_id: 3, tastemaker_item_type: 'track', tastemaker_item_owner_id: 2, action: 'repost', tastemaker_user_id: 10 },
+      user_ids: [10]
+    }
+    await insertNotifications(processor.discoveryDB, [notificationRow])
+
+    await insertMobileSettings(processor.identityDB, [
+      { userId: 10 },
+    ])
+    await insertMobileDevices(processor.identityDB, [
+      { userId: 10 },
+    ])
+    await new Promise((resolve) => setTimeout(resolve, 10))
```

Comment:
also check the test i linked earlier cause i was able to make it pass without this setTimeout, which imo we should try to avoid if possible.